### PR TITLE
 luasocket: fix non-5.1 builds

### PR DESCRIPTION
**Pull request by Mic92**
###### Motivation for this change

 LUAV has to be set accordingly for make,
 otherwise it will install files default to the 5.1 path.
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
The macos x test failed while building luaevent not luasocket.

**Comment by Mic92** [comment]
anything else to do?

**Comment by Mic92** [comment]
ping @flosse

File: pkgs/top-level/lua-packages.nix [hunk 1]
```diff
@@ -189,19 +189,23 @@ let
...
+        PLAT=${if stdenv.isDarwin then "macosx" else "linux"}
+        LUAPREFIX_linux=$out
+        LUAPREFIX_macosx=$out
+      );
+    '';
+
     meta = {
       homepage = "http://w3.impa.br/~diego/software/luasocket/";
-      hydraPlatforms = stdenv.lib.platforms.linux;
+      hydraPlatforms = stdenv.lib.platforms.all;
```
Comment (LnL7) [inline pkgs/top-level/lua-packages.nix]: This should probably be `platforms.unix` or just `platforms.linux ++ platforms.darwin`

Comment (Mic92) [inline pkgs/top-level/lua-packages.nix]: When to use platforms.all then? This library should build on unixoide platforms as well as windows.

Comment (LnL7) [inline pkgs/top-level/lua-packages.nix]: I just assumed the current expression does not work on other platforms because of line 200.

Comment (Mic92) [inline pkgs/top-level/lua-packages.nix]: this is a now more restricted list, which covers all platforms the makefile of luasocket targets.

**Comment by Mic92** [comment]
@zimbatm do you know, who usually maintains the lua stuff and could review this?

File: pkgs/top-level/lua-packages.nix [hunk 2]
```diff
@@ -197,15 +197,18 @@ let
...
+               else if stdenv.isSunOS then "solaris"
+               else throw "unsupported platform"}
+        prefix=$out
       );
     '';
 
     meta = {
       homepage = "http://w3.impa.br/~diego/software/luasocket/";
-      hydraPlatforms = stdenv.lib.platforms.linux;
+      hydraPlatforms = with stdenv.lib.platforms; [darwin linux freebsd illumos];
```
Comment (zimbatm) [inline pkgs/top-level/lua-packages.nix]: `with platforms;` is sufficient since `lib` is in scope (see line below).

**Comment by zimbatm** [comment]
@Mic92 not sure, there is quite a mix when looking at the git logs. I just have one nitpick but the changes look good to me.

**Comment by Mic92** [comment]
fixed

**Pull request merged**